bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java

```diff
@@ -333,13 +335,24 @@ public LedgerMetadata getLedgerMetadata() {
         return metadata;
     }
 
+    boolean setLedgerMetadata(LedgerMetadata expected, LedgerMetadata newMetadata) {
+        synchronized (this) {
+            if (metadata == expected) {
```

Comment:
add a comment ?
usually '==' seems an error in Java, but in this point we are doing it because it is expected

Reply:
I've never heard of '==' being code smell. This code is standard check and set. Can add a comment though.